packages-ts/gauntlet-terra-contracts/src/lib/contracts.ts

```diff
@@ -28,109 +29,164 @@ export type TerraABI = {
-  [CONTRACT_LIST.DEVIATION_FLAGGING_VALIDATOR]: 'deviation-flagging-validator',
-  [CONTRACT_LIST.OCR_2]: 'ocr2',
-  [CONTRACT_LIST.PROXY_OCR_2]: 'proxy-ocr2',
-  [CONTRACT_LIST.ACCESS_CONTROLLER]: 'access-controller',
-  [CONTRACT_LIST.CW20_BASE]: 'cw20_base',
-  [CONTRACT_LIST.CW4_GROUP]: 'cw4_group',
-  [CONTRACT_LIST.MULTISIG]: 'cw3_flex_multisig',
-}
+class ChainlinkContract extends Contract {
+  readonly downloadUrl = `https://github.com/smartcontractkit/chainlink-terra/releases/download/`
 
-const defaultContractVersions = {
-  [CONTRACT_LIST.FLAGS]: DEFAULT_RELEASE_VERSION,
-  [CONTRACT_LIST.DEVIATION_FLAGGING_VALIDATOR]: DEFAULT_RELEASE_VERSION,
-  [CONTRACT_LIST.OCR_2]: DEFAULT_RELEASE_VERSION,
-  [CONTRACT_LIST.ACCESS_CONTROLLER]: DEFAULT_RELEASE_VERSION,
-  [CONTRACT_LIST.CW20_BASE]: DEFAULT_CWPLUS_VERSION,
-  [CONTRACT_LIST.CW4_GROUP]: DEFAULT_CWPLUS_VERSION,
-  [CONTRACT_LIST.MULTISIG]: DEFAULT_CWPLUS_VERSION,
-}
-export const getContractABI = (contractId: CONTRACT_LIST): TerraABI => {
-  // Possible paths depending on how/where gauntlet is being executed
-  const possibleContractPaths = [
-    path.join(__dirname, './artifacts/contracts'),
-    path.join(process.cwd(), './contracts'),
-    path.join(process.cwd(), '../../contracts'),
-    path.join(process.cwd(), './packages-ts/gauntlet-terra-contracts/artifacts/contracts'),
-  ]
-
-  const toDirName = (contractId: CONTRACT_LIST) => contractDirName[contractId]
-
-  const abi = possibleContractPaths
-    .filter((path) => existsSync(`${path}/${toDirName(contractId)}/schema`))
-    .map((contractPath) => {
-      const toPath = (type) => {
-        if (contractId == CONTRACT_LIST.CW20_BASE && type == 'execute_msg') {
-          return path.join(contractPath, `./${toDirName(contractId)}/schema/cw20_${type}`)
-        } else {
-          return path.join(contractPath, `./${toDirName(contractId)}/schema/${type}`)
-        }
-      }
-      return {
-        execute: io.readJSON(toPath('execute_msg')),
-        query: io.readJSON(toPath('query_msg')),
-        instantiate: io.readJSON(toPath('instantiate_msg')),
-      }
-    })
-  if (abi.length === 0) {
-    logger.error(`ABI not found for contract ${contractId}`)
+  constructor(id, dirName, defaultVersion = DEFAULT_RELEASE_VERSION) {
+    super(id, dirName, defaultVersion)
   }
+}
+
+class CosmWasmContract extends Contract {
+  readonly downloadUrl = `https://github.com/CosmWasm/cw-plus/releases/download/`
 
-  return abi[0]
+  constructor(id, dirName, defaultVersion = DEFAULT_CWPLUS_VERSION) {
+    super(id, dirName, defaultVersion)
+  }
 }
 
-export const getContract = async (id: CONTRACT_LIST, version): Promise<Contract> => {
-  version = version ? version : defaultContractVersions[id]
-  // Preload contracts
-  return {
-    id,
-    abi: getContractABI(id),
-    bytecode: await getContractCode(id, version),
+class Contracts {
+  async getContract(id: CONTRACT_LIST, version: string): Promise<Contract> {
+    if (this[id]) {
```

Comment:
Store them into a contracts object (`this.contracts: { [key: string]: Contract }`), not directly into `this`